crates/oxc_transformer/src/plugins/replace_global_defines.rs

```diff
@@ -446,12 +476,16 @@ impl<'a> ReplaceGlobalDefines<'a> {
                         })
                     }
                     Expression::Identifier(ident) => {
-                        if !ident.is_global_reference(symbols) {
+                        if !ident.is_global_reference(ctx.symbols()) {
                             return false;
                         }
                         cur_part_name = &ident.name;
                         None
                     }
+                    Expression::ThisExpression(_) if should_replace_this_expr => {
+                        cur_part_name = &THIS_ATOM;
```

Comment:
`cur_part_name.as_str() == "this"` should do the same thing.

Reply:
Just want to make the code logic consistent, if use `cur_part_name.as_str() == "this"`, you need to check if the cursor is zero.